### ISSUE TITLE
Celery tag 5.0 specifies click>=7.0 and downloads click 8.0 now

### ISSUE THREAD

**Issue by Tomasz-Kluczkowski** [celery/celery#6768]
# Checklist

- [ ] I have verified that the issue exists against the `master` branch of Celery.
- [ ] This has already been asked to the [discussion group](https://groups.google.com/forum/#!forum/celery-users) first.
- [x] I have read the relevant section in the
  [contribution guide](http://docs.celeryproject.org/en/latest/contributing.html#other-bugs)
  on reporting bugs.
- [x] I have checked the [issues list](https://github.com/celery/celery/issues?q=is%3Aissue+label%3A%22Issue+Type%3A+Bug+Report%22+-label%3A%22Category%3A+Documentation%22)
  for similar or identical bug reports.
- [x] I have checked the [pull requests list](https://github.com/celery/celery/pulls?q=is%3Apr+label%3A%22PR+Type%3A+Bugfix%22+-label%3A%22Category%3A+Documentation%22)
  for existing proposed fixes.
- [x] I have checked the [commit log](https://github.com/celery/celery/commits/master)
  to find out if the bug was already fixed in the master branch.
- [x] I have included all related issues and possible duplicate issues
  in this issue (If there are none, check this box anyway).

## Mandatory Debugging Information

- [ ] I have included the output of ``celery -A proj report`` in the issue.
    (if you are not able to do this, then at least specify the Celery
     version affected). 5.0.0
- [ ] I have verified that the issue exists against the `master` branch of Celery.
- [x] I have included the contents of ``pip freeze`` in the issue.
- [x] I have included all the versions of all the external dependencies required
  to reproduce this bug.

## Optional Debugging Information

- [x] I have tried reproducing the issue on more than one Python version
  and/or implementation.
- [ ] I have tried reproducing the issue on more than one message broker and/or
  result backend.
- [ ] I have tried reproducing the issue on more than one version of the message
  broker and/or result backend.
- [ ] I have tried reproducing the issue on more than one operating system.
- [ ] I have tried reproducing the issue on more than one workers pool.
- [ ] I have tried reproducing the issue with autoscaling, retries,
  ETA/Countdown & rate limits disabled.
- [ ] I have tried reproducing the issue after downgrading
  and/or upgrading Celery and its dependencies.

#### Related Issues

- None

#### Possible Duplicates

- None

## Environment & Settings

**Celery version**: 5.0.0

<details>
<summary><b><code>celery report</code> Output:</b></summary>
<p>

```
```

</p>
</details>

# Steps to Reproduce
Try to install celery 5.0.0 now that click released version 8.0 - it will install click 8.0 and be broken since click_repl 0.1.6 does not work with click 8.0

When celery app is then started it produces this output:

```
/home/tomasz_kluczkowski/.virtualenvs/celery-experiments/bin/python /home/tomasz_kluczkowski/.virtualenvs/celery-experiments/bin/celery -A tasks worker -l INFO -E -c 4
Traceback (most recent call last):
  File "/home/tomasz_kluczkowski/.virtualenvs/celery-experiments/bin/celery", line 8, in <module>
    sys.exit(main())
  File "/home/tomasz_kluczkowski/.virtualenvs/celery-experiments/lib/python3.9/site-packages/celery/__main__.py", line 14, in main
    from celery.bin.celery import main as _main
  File "/home/tomasz_kluczkowski/.virtualenvs/celery-experiments/lib/python3.9/site-packages/celery/bin/celery.py", line 10, in <module>
    from celery.bin.amqp import amqp
  File "/home/tomasz_kluczkowski/.virtualenvs/celery-experiments/lib/python3.9/site-packages/celery/bin/amqp.py", line 7, in <module>
    from click_repl import register_repl
  File "/home/tomasz_kluczkowski/.virtualenvs/celery-experiments/lib/python3.9/site-packages/click_repl/__init__.py", line 6, in <module>
    import click._bashcomplete
ModuleNotFoundError: No module named 'click._bashcomplete'

Process finished with exit code 1
```

## Required Dependencies

* **Minimal Python Version**: I tried on 3.9, but it has no influence
* **Minimal Celery Version**: 5.0.0 exact
* **Minimal Kombu Version**: N/A 
* **Minimal Broker Version**: N/A 
* **Minimal Result Backend Version**: N/A 
* **Minimal OS and/or Kernel Version**: N/A 
* **Minimal Broker Client Version**: N/A 
* **Minimal Result Backend Client Version**: N/A 

### Python Packages

<details>
<summary><b><code>pip freeze</code> Output:</b></summary>
amqp==5.0.5
billiard==3.6.3.0
celery==5.0.0
click==8.0.0
click-didyoumean==0.0.3
click-plugins==1.1.1
click-repl==0.1.6
flower==1.0.1
humanize==0.5.1
kombu==5.0.2
prometheus-client==0.10.1
prompt-toolkit==3.0.17
pytz==2021.1
redis==3.5.3
six==1.15.0
tornado==6.1
vine==5.0.0
wcwidth==0.2.5

<p>

```
```

</p>
</details>

### Other Dependencies

<details>
<p>
N/A
</p>
</details>

## Minimally Reproducible Test Case

<details>
pip uninstall celery click
pip install celery==5

this will now pull celery 5 plus click 8.0 (and all the other stuff celery needs, only click 8.0 is to be fixed)
<p>

```python
```

</p>
</details>

# Expected Behavior

celery requires click version which can actually be used with click_repl

# Actual Behavior
exception on starting celery app

Starting celery 5.0.0 with click 8.0 produces an exception as below:

```
/home/tomasz_kluczkowski/.virtualenvs/celery-experiments/bin/python /home/tomasz_kluczkowski/.virtualenvs/celery-experiments/bin/celery -A tasks worker -l INFO -E -c 4
Traceback (most recent call last):
  File "/home/tomasz_kluczkowski/.virtualenvs/celery-experiments/bin/celery", line 8, in <module>
    sys.exit(main())
  File "/home/tomasz_kluczkowski/.virtualenvs/celery-experiments/lib/python3.9/site-packages/celery/__main__.py", line 14, in main
    from celery.bin.celery import main as _main
  File "/home/tomasz_kluczkowski/.virtualenvs/celery-experiments/lib/python3.9/site-packages/celery/bin/celery.py", line 10, in <module>
    from celery.bin.amqp import amqp
  File "/home/tomasz_kluczkowski/.virtualenvs/celery-experiments/lib/python3.9/site-packages/celery/bin/amqp.py", line 7, in <module>
    from click_repl import register_repl
  File "/home/tomasz_kluczkowski/.virtualenvs/celery-experiments/lib/python3.9/site-packages/click_repl/__init__.py", line 6, in <module>
    import click._bashcomplete
ModuleNotFoundError: No module named 'click._bashcomplete'

Process finished with exit code 1
```
This can be easily fixed by changing requirements for 5.0.0 tag for click to be same as on 5.0.5: click >=7, <8

**Comment by thedrow** [comment]
Yes, we should limit the Click version for now until we resolve this issue.

**Comment by maybe-sybr** [comment]
Hi @Tomasz-Kluczkowski -- looks like this was fixed in 5.0.3 (specifically in 208e90e). I don't think we're too likely to backport a fix onto the 5.0.0/.1/.2 tags since we'd need to do post releases for that which AFAICT we haven't done before. Is upgrading to `>=5.0.3,<5.1` an option for you?

**Comment by thedrow** [comment]
I was sure we fixed this. Thanks, @maybe-sybr!
